File: example/bench_struct_method.py
class Counter:
    __slots__ = ("val",)

    def __init__(self):
        self.val = 0

    def inc(self):
        self.val = self.val + 1

    def add(self, amount):
        self.val = self.val + amount

    def get(self):
        return self.val


def run(n):
    c = Counter()
    i = 0
    while i < n:
        c.inc()
        c.add(i)
        i = i + 1
    return c.get()


run(500_000)

File: example/bench_struct_method.ts
class Counter {
    val: number = 0;

    inc(): void {
        this.val = this.val + 1;
    }

    add(amount: number): void {
        this.val = this.val + amount;
    }

    get(): number {
        return this.val;
    }
}

function run(n: number): number {
    const c = new Counter();
    let i = 0;
    while (i < n) {
        c.inc();
        c.add(i);
        i = i + 1;
    }
    return c.get();
}

run(500_000);
